io/eolearn/io/processing_api.py

```diff
@@ -6,84 +6,51 @@
 
 LOGGER = logging.getLogger(__name__)
 
 
-class SentinelHubInputTask(EOTask):
-    """ A processing API input task that loads 16bit integer data and converts it to a 32bit float feature.
-    """
-    def __init__(self, data_source, size=None, resolution=None, bands_feature=None, bands=None, additional_data=None,
-                 maxcc=1.0, time_difference=None, cache_folder=None, max_threads=None, config=None,
-                 bands_dtype=np.float32):
+class SentinelHubInputBase(EOTask):
+    ''' Base class for Processing API input tasks
+    '''
+    def __init__(self, data_source, size=None, resolution=None, cache_folder=None, config=None, max_threads=None):
         """
         :param data_source: Source of requested satellite data.
         :type data_source: DataSource
         :param size: Number of pixels in x and y dimension.
         :type size: tuple(int, int)
         :type resolution: Resolution in meters, passed as a tuple for X and Y axis.
         :type resolution: tuple(int, int)
-        :param bands_feature: Target feature into which to save the downloaded images.
-        :type bands_feature: tuple(sentinelhub.FeatureType, str)
-        :param bands: An array of band names.
-        :type bands: list[str]
-        :param additional_data: A list of additional data to be downloaded, such as SCL, SNW, dataMask, etc.
-        :type additional_data: list[tuple(sentinelhub.FeatureType, str)]
-        :param maxcc: Maximum cloud coverage.
-        :type maxcc: float
-        :param time_difference: Minimum allowed time difference, used when filtering dates, None by default.
-        :type time_difference: datetime.timedelta
         :param cache_folder: Path to cache_folder. If set to None (default) requests will not be cached.
         :type cache_folder: str
+        :param config: An instance of SHConfig defining the service
+        :type config: SHConfig or None
         :param max_threads: Maximum threads to be used when downloading data.
         :type max_threads: int
         """
+
         self.size = size
         self.resolution = resolution
-        self.data_source = data_source
-        self.maxcc = maxcc
-        self.time_difference = dt.timedelta(seconds=1) if time_difference is None else time_difference
-        self.cache_folder = cache_folder
-        self.max_threads = max_threads
         self.config = config or SHConfig()
-        self.bands_dtype = bands_dtype
-
-        self.bands_feature = next(self._parse_features(bands_feature)()) if bands_feature else None
-
-        if bands is not None:
-            self.bands = bands
-        elif bands_feature is not None:
-            self.bands = data_source.bands()
-        else:
-            self.bands = []
-
-        if additional_data is None:
-            self.additional_data = []
-        else:
-            self.additional_data = list(self._parse_features(additional_data, new_names=True)())
+        self.timestamp = None
```

Comment:
If I'm not mistaken `self.timestamp` is not used anywhere and could be removed.

Reply:
Done.